server/models/File/FileMutation.js

```diff
@@ -25,24 +26,59 @@ export const fileUploadHandler = async (req, res) => {
     } = req.file;
 
     try {
-      const query = r.db('work_genius').table('files').insert({
+      const createdAt = new Date().getTime();
+      const fileMetadata = {
         id: filename,
         type: mimetype,
         name: originalname,
-      });
+        url: `${SERVER_FILES_URL}/${filename}`,
+        path: `${targetDir}/${filename}`,
+        createdAt
+      };
+      const query = r.db('work_genius').table('files').insert(fileMetadata);
       const connection = await r.connect({ host: DB_HOST, port: DB_PORT });
       await query.run(connection);
       await connection.close();
-      res.status(200).send({
-        id: filename,
-        type: mimetype,
-        name: originalname,
-        url: `${SERVER_FILES_URL}/${filename}`
-      });
+      res.status(200).send(fileMetadata);
     } catch (dbErr) {
       res.status(dbErr.status).send({
         error: dbErr
       });
     }
   });
 };
+
+export const fileDeleteHandler = async (req, res) => {
+  try {
+    const fileId = req.params.id;
+    const connection = await r.connect({ host: DB_HOST, port: DB_PORT });
+    let query = null, result = null;
+    query = r.db('work_genius')
+      .table('files')
+      .get(fileId);
+    result = await query.run(connection);
+
+    fs.unlink(result.path, async (deleteErr) => {
+      if (deleteErr) {
+      // not sure why catch didn't handle this throw deleteError
```

Comment:
Please refer to this article
http://syshen.cc/post/10726736762/catch-exception-thrown-in-javascript-callback